jbpm-console-ng-case-mgmt/jbpm-console-ng-case-mgmt-backend/src/test/java/org/jbpm/console/ng/cm/backend/server/CaseInstanceMapperTest.java

```diff
@@ -36,19 +40,29 @@ public static void assertCaseInstance(final CaseInstance ci, final CaseInstanceS
         assertEquals(ci.getStartedAt(), cis.getStartedAt());
         assertEquals(ci.getCompletedAt(), cis.getCompletedAt());
         assertEquals(ci.getCaseDefinitionId(), cis.getCaseDefinitionId());
+        assertCaseStages(ci.getStages(), cis.getStages());
+    }
+
+    public static void assertCaseStages(final List<CaseStage> csl, final List<CaseStageSummary> cssl) {
+        assertNotNull(cssl);
+        int i = 0;
+        CaseStage caseStage;
+        for (CaseStageSummary caseStageSummary : cssl) {
```

Comment:
Mixing for-each loop + explicit index looks confusing. Why not use standard for loop for comparison like:

    assertTrue(csl.size() == cssl.size());
    for (int i = 0; i < csl.size(); i++) {
     CaseStage original = csl.get(i);
     CaseStageSummary copied = cssl.get(i);
     assertEquals(...);
    }
Also I'd add assert that verifies that both lists have the same size.